k4FWCore/include/k4FWCore/ICollectionFromObjectSvc.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2014-2024 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef FWCORE_ICOLLECTIONFROMOBJECTSVC_H
+#define FWCORE_ICOLLECTIONFROMOBJECTSVC_H
+
+#include <GaudiKernel/IInterface.h>
+
+#include <podio/CollectionBase.h>
+#include <podio/ObjectID.h>
+#include <podio/utilities/TypeHelpers.h>
+
+#include <fmt/format.h>
+
+class ICollectionFromObjectSvc : virtual public IInterface {
+public:
+  DeclareInterfaceID(ICollectionFromObjectSvc, 1, 0);
+
+  template <podio::ObjectType O>
```

Comment:
This could also be changed to something that works with podio 1.3, as this is not part of that tag.